### PR TITLE
SDC questionnaire integer edit text widget

### PR DESCRIPTION
**Pull request by kashyapj2793**
Addressing #140

### CODE REVIEW

File: datacapture/src/androidTest/java/com/google/android/fhir/datacapture/views/QuestionnaireItemEditTextIntegerViewHolderFactoryInstrumentedTest.kt [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        QuestionnaireResponse.QuestionnaireResponseItemComponent()
+      ))
+
+    assertThat(viewHolder.itemView.findViewById<TextInputEditText>(R.id.textInputEditText).hint)
+      .isEqualTo("Question?")
+  }
+
+  @Test
+  @UiThreadTest
+  fun shouldSetInputText() {
```
Comment (jingtang10) [inline datacapture/src/androidTest/java/com/google/android/fhir/datacapture/views/QuestionnaireItemEditTextIntegerViewHolderFactoryInstrumentedTest.kt:66]: If there isn't an answer already, I think the edit text should remain empty.

File: datacapture/src/androidTest/java/com/google/android/fhir/datacapture/views/QuestionnaireItemEditTextIntegerViewHolderFactoryInstrumentedTest.kt [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        }
+      ))
+
+    assertThat(viewHolder.itemView.findViewById<TextInputEditText>(R.id.textInputEditText)
+      .text.toString().toIntOrNull()).isEqualTo(0)
+  }
+
+  @Test
+  @UiThreadTest
+  fun shouldSetQuestionnaireResponseItemComponentAnswer() {
```
Comment (jingtang10) [inline datacapture/src/androidTest/java/com/google/android/fhir/datacapture/views/QuestionnaireItemEditTextIntegerViewHolderFactoryInstrumentedTest.kt:85]: add a test case for if the edit text content is empty?

File: datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemEditTextIntegerViewHolderFactory.kt [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+          private lateinit var textInputEditText: TextInputEditText
+          private lateinit var questionnaireItemViewItem: QuestionnaireItemViewItem
+
+          override fun init(itemView: View) {
+              textInputLayout = itemView.findViewById(R.id.textInputLayout)
+              textInputEditText = itemView.findViewById(R.id.textInputEditText)
+              textInputEditText.setRawInputType(InputType.TYPE_CLASS_NUMBER or
+                InputType.TYPE_NUMBER_FLAG_SIGNED)
+              textInputEditText.doAfterTextChanged { editable: Editable? ->
+                  questionnaireItemViewItem.singleAnswerOrNull =
```
Comment (jingtang10) [inline datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemEditTextIntegerViewHolderFactory.kt:44]: in the case where there isn't any input (text box empty), we probably should make sure to remove any answer.

**Review by jingtang10 — CHANGES_REQUESTED** [review]
Thanks for the change @kashyapj2793!

File: datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemEditTextIntegerViewHolderFactory.kt (hunk 3 above)
Comment (kashyapj2793) [reply datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemEditTextIntegerViewHolderFactory.kt:44]: If there isn't any input questionnaireItemViewItem.singleAnswerOrNull = null since  editable.toString().toIntOrNull() will be null

File: datacapture/src/androidTest/java/com/google/android/fhir/datacapture/views/QuestionnaireItemEditTextIntegerViewHolderFactoryInstrumentedTest.kt (hunk 2 above)
Comment (kashyapj2793) [reply datacapture/src/androidTest/java/com/google/android/fhir/datacapture/views/QuestionnaireItemEditTextIntegerViewHolderFactoryInstrumentedTest.kt:85]: done

File: datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemEditTextIntegerViewHolderFactory.kt (hunk 3 above)
Comment (kashyapj2793) [reply datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemEditTextIntegerViewHolderFactory.kt]: I think I have done this, please check the latest commit

File: datacapture/src/androidTest/java/com/google/android/fhir/datacapture/views/QuestionnaireItemEditTextIntegerViewHolderFactoryInstrumentedTest.kt (hunk 1 above)
Comment (kashyapj2793) [reply datacapture/src/androidTest/java/com/google/android/fhir/datacapture/views/QuestionnaireItemEditTextIntegerViewHolderFactoryInstrumentedTest.kt:66]: This is checking if there is an answer already the edit text should be set correctly

Comment (jingtang10) [reply datacapture/src/androidTest/java/com/google/android/fhir/datacapture/views/QuestionnaireItemEditTextIntegerViewHolderFactoryInstrumentedTest.kt:66]: add another test case where questionnaire response doesn't contain an answer, and check the input is an empty string.

**Review by jingtang10 — APPROVED** [review]
thanks @kashyapj2793!

just one comment in the test file. otherwise good to merge.
